test_package.go

```diff
@@ -20,6 +20,13 @@ import (
 
 // Deployment provides a way for tests to interact with a set of homeservers.
 type Deployment interface {
+	// Returns the resolvable server name (host) of a homeserver given its short alias
+	// (e.g., "hs1", "hs2").
+	//
+	// In the case of the standard Docker deployment, this will be the same `hs1`, `hs2`
+	// but may be different for other custom deployments (ex.
+	// `shardDeployment1.GetFullyQualifiedHomeserverName(t, "hs1")` -> `hs1.shard1:8081`).
+	GetFullyQualifiedHomeserverName(t ct.TestLike, hsName string) string
```

Comment:
@kegsay Just to double-check with the Complement taste-maker, do these changes make sense to you?

Are you okay with this breaking change to the `complement.Deployment` interface?

Are you okay with the breaking changes to `complement.WithDeployment(...)` / `complement.WithCleanup(...)` changes?

Reply:
I'm happy with adding more params to `With...` options.

I'm unhappy with adding `GetFullyQualifiedHomeserverName`. See https://github.com/matrix-org/complement/pull/778/files#r2091384008

Reply:
Thanks for the review @kegsay!

The `GetFullyQualifiedHomeserverName` changes have been split off to https://github.com/matrix-org/complement/pull/780